### PR TITLE
Reduce explicit use of default definition scope

### PR DESCRIPTION
**Pull request by SkySkimmer**
(renamed Locality.locality -> Locality.definition_scope)

Overlays:
- https://github.com/mattam82/Coq-Equations/pull/450

### CODE REVIEW

File: vernac/comDefinition.mli [hunk 1]
```diff
@@ -28,7 +28,7 @@ val interp_definition
 val do_definition
   :  ?hook:Declare.Hook.t
   -> name:Id.t
-  -> scope:Locality.locality
+  -> ?scope:Locality.definition_scope
```
Comment (gares) [inline vernac/comDefinition.mli:31]: please document the default value

File: vernac/locality.mli [hunk 2]
```diff
@@ -11,7 +11,9 @@
 (** * Managing locality *)
 
 type import_status = ImportDefaultBehavior | ImportNeedQualified
-type locality = Discharge | Global of import_status
+type definition_scope = Discharge | Global of import_status
+
+val default_scope : definition_scope
```
Comment (gares) [inline vernac/locality.mli:16]: here

Comment (gares) [inline vernac/locality.mli:16]: here

Comment (SkySkimmer) [reply vernac/locality.mli:16]: updated

**Review by Alizter — APPROVED** [review]

**Comment by Alizter** [comment]
merging this soon

**Comment by Alizter** [comment]
@coqbot merge now

**Pull request merged**